ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -11767,40 +11767,19 @@ void HqlCppTranslator::buildScriptFunctionDefinition(BuildCtx &funcctx, IHqlExpr
     createParam.append(isImport ? "EFimport" : "EFembed");
     if (returnType->getTypeCode()==type_void)
         createParam.append("|EFnoreturn");
-    if (formals->numChildren()==0)
-        createParam.append("|EFnoparams");
 
-    HqlExprArray attrArgs;
-    ForEachChild(idx, bodyCode)
+    IHqlExpression *optionsParam = nullptr;
+    if (formals->numChildren())
     {
-        IHqlExpression *child = bodyCode->queryChild(idx);
-        if (child->isAttribute() && child->queryName() != languageAtom && child->queryName() != importAtom)
-        {
-            StringBuffer attrParam;
-            if (attrArgs.ordinality())
-                attrParam.append(",");
-            attrParam.append(child->queryName());
-
-            IHqlExpression * value = child->queryChild(0);
-            if (value)
-                attrParam.append("=");
-            attrArgs.append(*createConstant(attrParam));
-            if (value)
-                attrArgs.append(*ensureExprType(value, unknownStringType));
-        }
+        optionsParam = formals->queryChild(formals->numChildren()-1);
+        assertex(streq(str(optionsParam->queryId()), "__options"));
```

Comment:
Normally __optionsId would be added to hqlcatom.?pp, this test would become a pointer comparison, and there would be no createIdAtom() call below.